rules/opentitan_test.bzl

```diff
@@ -310,17 +310,20 @@ def opentitan_functest(
         elif target in ["cw310_rom", "cw310_test_rom"]:
             devices_to_build_for.append("fpga_cw310")
 
+            # If the user passed in a bitstream, record it here so that we do
+            # not override it later
+            user_bitstream = cw310.get("bitstream", None) if cw310 else None
             # Copy `cw310` for each `target`. This is not a deep copy, thus we
             # also copy tags below.
             cw310_ = cw310_params() if not cw310 else dict(cw310.items())
             if "test_rom" in target:
                 cw310_["rom_kind"] = "testrom"
-                cw310_["bitstream"] = "@//hw/bitstream:test_rom"
+                cw310_["bitstream"] = user_bitstream if user_bitstream else "@//hw/bitstream:test_rom"
```

Comment:
It would be more concise to say `user_bitstream or "@//hw/bitstream:test_rom"`, but honestly I think this is clearer.

Reply:
🤯

I did not know `or` worked like that in python. I think I might leave it as is to keep the intent clear.

Reply:
The `if` construct used here is the correct way to express this sort of conditional (ie: where you'd use `: ?` in C).

The logical `and` and `or` operators can get squirrely when you make an assumption about the types of values on the left and right sides of a such an expression.  Suddenly a user supplies a different type that gets evaluated as truthy or falsey causing the overall result to be the wrong type.

See also commandments 3 and 5 of [The 10 Commandments for C programmers](https://www.lysator.liu.se/c/ten-commandments.html).

Reply:
Huh, I'm having trouble imagining values/types of `a` and `b` such that `a if a else b` ≠ `a or b`. Or did I miss your point?

Reply:
(Just curious. I actually prefer the explicit `if` here for clarity.)

Reply:
> burned more than once by python doing something unexpected

Not directly related to this case but this comment brought back memories: [PEP 238 – Changing the Division Operator](https://peps.python.org/pep-0238/)